pkg/runtime/serializer/codec_factory.go

```diff
@@ -308,3 +310,52 @@ func (f CodecFactory) SerializerForFileExtension(extension string) (runtime.Seri
 	}
 	return nil, false
 }
+
+type DirectCodecFactory struct {
+	CodecFactory
+}
+
+func (f DirectCodecFactory) EncoderForVersion(serializer runtime.Encoder, gv unversioned.GroupVersion) runtime.Encoder {
+	return DirectCodec{
+		struct {
```

Comment:
I would not use anonymous structs.  Also, there should be a struct for this already in `pkg/runtime/codecs.go`